src/main/java/net/minecraftforge/common/extensions/IForgePacketBuffer.java

```diff
@@ -0,0 +1,109 @@
+package net.minecraftforge.common.extensions;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistry;
+import net.minecraftforge.registries.IForgeRegistry;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+import net.minecraftforge.registries.RegistryManager;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * Extension-Interface providing methods for writing registry-id's instead of their registry-names.
```

Comment:
Could this be better served by a static utility class instead?
I am not sure of the benefit of making this an interface.

Reply:
It's much more discoverable since the methods are directly available on the vanilla type. This is the new pattern we are using for patching in methods.